Snowflake.Data/Core/FileTransfer/EncryptionProvider.cs

```diff
@@ -124,8 +124,15 @@ static private byte[] encryptFileKey(byte[] masterKey, byte[] unencryptedFileKey
             MemoryStream targetStream = new MemoryStream();
             CryptoStream cryptoStream = new CryptoStream(targetStream, aes.CreateEncryptor(), CryptoStreamMode.Write);
             
-            byte[] inFileBytes = File.ReadAllBytes(inFile);
-            cryptoStream.Write(inFileBytes, 0, inFileBytes.Length);
+            using(Stream inStream = File.OpenRead(inFile))
+            {
+                byte[] buffer = new byte[2048];
```

Comment:
I would like to see more performance study about this buffer size: will there be difference when the buffer size change from like, 1K to 16M? What should be default buffer size?

Reply:
I ran tests using different buffer sizes on a 0.6 MB, 60 MB, and 600 MB file and here are the results:

 
|     Buffer size   |     0.6 MB    |      60 MB    |     600 MB    |
| ------ | ------------- | ------------- | ------------- |
|   1 KB | 00.0077772 s  | 00.3869570 s  | 04.3909098 s  |
|   4 KB | 00.0071792 s  | 00.3684152 s  | 03.6724831 s  |
|  16 KB | 00.0065944 s  | 00.2287245 s  | 02.5791550 s  |
|  32 KB | 00.0054242 s  | 00.2070174 s  | 02.2068000 s  |
|  64 KB | 00.0063405 s  | 00.1965125 s  | 02.1274505 s  |
| 128 KB | 00.0046953 s  | 00.1906225 s  | **02.0630387 s**  |
| 256 KB | **00.0044551 s**  | **00.1892029 s**  | 02.0714631 s  |
| 512 KB | 00.0076899 s  | 00.1899127 s  | 02.1211855 s  |
|   1 MB | 00.0082933 s  | 00.1922759 s  | 02.0771062 s  |
|   8 MB | 00.0110063 s  | 00.2204441 s  | 02.4287913 s  |
|  16 MB | 00.0146081 s  | 00.2142795 s  | 02.5212195 s  |


The buffer range that performed fastest was 128 KB - 256 KB. Should the default be 128 KB?